tests/sentry/api/endpoints/test_project_metrics_extraction_rules.py

```diff
@@ -772,3 +775,137 @@ def test_query_filter_rules(self):
         data = response.data
         assert len(data) == 1
         assert {el["spanAttribute"] for el in data} == {"some_span"}
+
+    @django_db_all
+    @with_feature("organizations:custom-metrics-extraction-rule")
+    def test_alerts_and_widgets_are_deleted_with_extraction_rules(self):
+        config = {
+            "spanAttribute": "count_clicks",
+            "aggregates": ["count", "p50", "p75", "p95", "p99"],
+            "unit": "none",
+            "tags": ["tag1", "tag2"],
+            "conditions": [
+                {"value": "foo:bar"},
+            ],
+        }
+        span_attribute_extraction_rule = self.create_span_attribute_extraction_config(
+            dictionary=config, user_id=self.user.id, project=self.project
+        )
+        condition1 = span_attribute_extraction_rule.conditions.first()
+        condition_mris = condition1.generate_mris()
+        mri1 = condition_mris[0]
+        mri2 = condition_mris[1]
+        alert_rule1 = self.create_alert_rule(organization=self.organization, aggregate=mri1)
+        alert_rule2 = self.create_alert_rule(organization=self.organization, aggregate=mri2)
+        dashboard = self.create_dashboard(organization=self.organization)
+        dashboard_widget1 = self.create_dashboard_widget(dashboard=dashboard, order=0)
+        self.create_dashboard_widget_query(widget=dashboard_widget1, aggregates=[mri1], order=1)
+        dashboard_widget2 = self.create_dashboard_widget(dashboard=dashboard, order=2)
+        self.create_dashboard_widget_query(widget=dashboard_widget2, aggregates=[mri2], order=2)
+
+        updated_rule = {
+            "metricsExtractionRules": [
+                {
+                    "spanAttribute": "count_clicks",
+                    "aggregates": ["count"],
+                    "unit": "none",
+                    "tags": ["tag1", "tag2", "tag3"],
+                    "conditions": [
+                        {"id": 1, "value": "other:condition"},
+                    ],
+                }
+            ]
+        }
+
+        self.get_success_response(
+            self.organization.slug,
+            self.project.slug,
+            method="delete",
+            **updated_rule,
+        )
+
+        assert not SpanAttributeExtractionRuleConfig.objects.filter(
+            project=self.project, span_attribute="count_clicks"
+        ).exists()
+
+        assert not AlertRule.objects.filter(id=alert_rule1.id).exists()
+        assert not AlertRule.objects.filter(id=alert_rule2.id).exists()
+        assert not DashboardWidget.objects.filter(id=dashboard_widget1.id).exists()
+        assert not DashboardWidget.objects.filter(id=dashboard_widget2.id).exists()
+
+    @django_db_all
+    @with_feature("organizations:custom-metrics-extraction-rule")
+    def test_alerts_and_widgets_connected_to_other_extraction_rules_are_not_affected(self):
```

Comment:
One of these tests verifies correct deletion and the other verifies unaffected other extraction rules. Could you add another test here that verifies that when changing an extraction rule, only those rules are deleted that the specific combination of spanAttribute, aggregate and condition is targeting?